src/main/java/hudson/remoting/Engine.java

```diff
@@ -441,6 +410,12 @@ private Socket connect(String host, String port) throws IOException, Interrupted
     private void waitForServerToBack() throws InterruptedException {
         Thread t = Thread.currentThread();
         String oldName = t.getName();
+        SSLSocketFactory sslSocketFactory = null;
+        try {
+            sslSocketFactory = getSSLSocketFactory();
+        } catch (Throwable e) {
```

Comment:
This approach is being used in other places of this file. It is not LGTM, but it does not increase the technical debt